lib/PGObject/Util/DBAdmin.pm

```diff
@@ -559,26 +559,62 @@ sub server_version {
 }
 
 
-=head2 list_dbs([$dbname])
+=head2 list_dbs([$dbname],[$excludes])
 
 Returns a list of db names.
 
 When a database name is specified, uses that database to connect to,
 using the credentials specified in the instance.
 
 If no database name is specified, 'template1' is used.
+If an array of excluded databases is provided, it is used to limit the list
```

Comment:
Ok. But what is the reason we want this? I mean Perl has a nice construct with `grep { not $excludes{$_} } @all_databases` where you can really simply limit the values in a list. Is there a reason this *must* happen in the SQL phase?

Reply:
The latest commit does address the SQL injection, thanks. That leaves the above question (why is it necessary to do this at the query level) though.